### PR TITLE
Upgrade eslint to v4

### PR DESCRIPTION
**Pull request by jasonnutter**
Updating eslint to v4 along with other eslint deps.

### CODE REVIEW

File: rules/base.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        "no-new-func": 2,
+        "no-new-object": 2,
+        "no-new-wrappers": 2,
+        "no-obj-calls": 2,
+        "no-octal": 2,
+        "no-octal-escape": 2,
+        "no-param-reassign": 2,
+        "no-proto": 2,
+        "no-redeclare": 2,
+        "no-return-assign": 0,
```
Comment (jasonnutter) [inline rules/base.json:81]: This is new, to allow more concise syntax when using refs:

```jsx
ref={input => this.input = input}
```

File: rules/base.json [hunk 2]
```diff
@@ -0,0 +1,116 @@
+{
+    // "$schema": "http://json.schemastore.org/eslintrc",
```
Comment (jasonnutter) [inline rules/base.json:2]: Uncomment while working on these files to get some intellisense in VS Code.

File: rules/react.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    "plugins": [
+        "react"
+    ],
+    "parserOptions": {
+        "jsx": true
+    },
+    "rules": {
+        "jsx-quotes": [ 2, "prefer-double" ],
+        "react/display-name": 2,
+        "react/forbid-prop-types": 1,
```
Comment (jasonnutter) [inline rules/react.json:12]: This is new, just a warning for now.

**Comment by pwmckenna** [comment]
lgtm!

**Pull request merged**